### PR TITLE
[5.5] add a setting for downloadable links

### PR DESCRIPTION
**Pull request by naneri**
make a user to be able to choose whether a link is downloadable. This is usefull when distributing files from 3rd party services and cant change the response to download response.

### CODE REVIEW

File: src/Illuminate/Mail/resources/views/html/button.blade.php [hunk 1]
```diff
@@ -7,7 +7,7 @@
                         <table border="0" cellpadding="0" cellspacing="0">
                             <tr>
                                 <td>
-                                    <a href="{{ $url }}" class="button button-{{ $color or 'blue' }}" target="_blank">{{ $slot }}</a>
+                                    <a href="{{ $url }}" class="button button-{{ $color or 'blue' }}" @if($donwload) download @endif target="_blank">{{ $slot }}</a>
```
Comment (jmarcher) [inline src/Illuminate/Mail/resources/views/html/button.blade.php]: Typo here

**Comment by naneri** [comment]
@jmarcher fixed. Thanks

**Comment by jmarcher** [comment]
This will break if I do not have a download variable in my view.
And I do not see any benefits from this, you could just extend the view and modify it

**Comment by naneri** [comment]
@jmarcher I can make a check if the `$download ` var isset and is true. Concerning the benefit - my current usecase is - I do have a CSV stored to S3 bucket. When I return the download link in the email, it just opens the CSV in new tab in Chrome. I would like to make it donwloadable directly.

**Comment by jmarcher** [comment]
Sure but that is just your use case if you want to modify the views you are free to do it by publishing them.

And btw, I never expect to directly download something from an email if it is not attached.

**Comment by taylorotwell** [comment]
No plans to add right now.

**Comment by naneri** [comment]
@jmarcher concerning the attachment - files are larger than 25mb in my case.

**Pull request closed without merging**